posthog/warehouse/data_load/service.py

```diff
@@ -64,8 +119,14 @@ def get_sync_schedule(external_data_schema: "ExternalDataSchema"):
             ),
         ),
         spec=ScheduleSpec(
-            intervals=[ScheduleIntervalSpec(every=sync_frequency)],
-            jitter=jitter,
+            calendars=[
+                ScheduleCalendarSpec(
+                    hour=[ScheduleRange(start=hour_of_day, end=end_hour, step=step)],
+                    minute=[ScheduleRange(start=minute_of_hour, end=minute_of_hour, step=1)],
+                )
+            ],
+            # No jitter when it's not midnight
+            jitter=jitter if minute_of_hour == 0 and hour_of_day == 0 else None,
```

Comment:
Can we get some tests on this to ensure the calendar spec is exactly what we expect given a bunch of different "first sync time" and "sync frequencies" please